plutus-core/plutus-ir/test/TransformSpec.hs

```diff
@@ -222,6 +238,11 @@ inline =
             , "letNonPureMultiStrict"
             ]
 
+-- | Check whether a term is globally unique.
+checkUniques :: (Ord a, MonadError (UniqueError a) m) => Term TyName Name uni fun a -> m ()
+checkUniques =
+    Uniques.checkTerm (\case { FreeVariable{} -> False; IncoherentUsage {} -> False; _ -> True})
```

Comment:
I think we should also be checking for `MultiplyDefined`, since I think locally but not globally unique programs will trigger that. I think `propRenameCheck` should probably do that too.

Reply:
Right now, we only throw when the error is `MultiplyDefined`. For checking uniqueness, this is right? Do we want to check everything/more things in this check? I guess why not? We want to make sure the inliner doesn't mess things up in other ways too.

Reply:
Argh, my bad, I had the meaning of the predicate backwards!

Reply:
OK I turned on throwing the free variables error and got really strange results. See [here](https://gist.githubusercontent.com/thealmarty/bc16f24a4e45e60db06d0273d3224c8a/raw/19ff15342f381b344ff4bc12d0f802b3c9ea2e28/freeVariableInliner.md). Maybe we can investigate in another ticket. For this PR I'll only throw `MultiplyDefined`.

Reply:
Probably because the uniques checker is going to be totally wrong because of https://github.com/input-output-hk/plutus/pull/5251#discussion_r1165314284

Reply:
ah good point! I'm working on that now.

Reply:
so do you think I should throw the other errors?

Reply:
No, it's correct as it is. We want to check for global uniqueness, that's what it does.

Reply:
Yes, sure. We have another ticket for adding other checks to the inliner I believe.